CONTRIBUTING: Explain reasons for track dependencies on this repo (#1406)
exercism/problem-specifications: CONTRIBUTING.md

* Do the same as when [porting an
  exercise](#porting-an-exercise-to-another-language-track).
  Reference the PR in problem-specifications.

>   It's suggested but not required to wait until the problem-specifications PR is merged before merging the track-specific PR, for the following reasons:

Reviewer: It's suggested, but not required, to wait ......

Author: the commas, right? okay

>     * If changes are suggested to the problem-specifications PR, it's highly likely that they will be applicable to the track-specific PR as well.

Reviewer: ```suggestion
    * If changes are suggested to the problem-specifications PR, it is likely that they will be applicable to the track-specific PR as well.
```

>     * Only applicable if the exercise needs a custom `title` in metadata.yml (as described in https://github.com/exercism/docs/blob/master/you-can-help/make-up-new-exercises.md#problem-specification): The title of the exercise as displayed on the website will use the default algorithm instead of the correct title.

Author: https://github.com/exercism/problem-specifications/pull/901

>     * Only applicable to tracks that have specific tools that depend on exercises being present in problem-specifications (see the track-specific documentation for whether any such tools exist): Such tools may operate unpredictably if the exercise does not exist in problem-specifications yet. Try checking out the branch on a local copy of problem-specifications or rerunning the tool after the problem-specifications PR is merged if applicable.

Author: as you can see, it is my attempt to make the https://github.com/exercism/problem-specifications/pull/1384 general. I definitely think track-specific tools deserve a mention, though exactly *what* the tools are has to be documented by the individual tracks.

>     * Only applicable to tracks that have specific tools that depend on exercises being present in problem-specifications (see the track-specific documentation for whether any such tools exist): Such tools may operate unexpectedly if the exercise does not exist in problem-specifications yet. Try checking out the branch on a local copy of problem-specifications or rerunning the tool after the problem-specifications PR is merged if applicable.

Reviewer: ```suggestion
    * Only applicable to tracks that have specific tools that depend on exercises being present in problem-specifications (see the track-specific documentation for whether any such tools exist): Such tools may operate unexpectedly if the exercise does not yet exist in problem-specifications. Try checking out the branch on a local copy of problem-specifications or rerunning the tool after the problem-specifications PR is merged if applicable.
```
